src/app/public/modules/datepicker/daypicker-cell.component.ts

```diff
@@ -0,0 +1,178 @@
+import {
+  ChangeDetectionStrategy,
+  Component,
+  Input,
+  OnDestroy,
+  OnInit
+} from '@angular/core';
+
+import {
+  SkyPopoverMessage,
+  SkyPopoverMessageType
+} from '@skyux/popovers';
+
+import {
+  takeUntil
+} from 'rxjs/operators';
+
+import {
+  Subject
+} from 'rxjs';
+
+import {
+  SkyDatepickerCalendarInnerComponent
+} from './datepicker-calendar-inner.component';
+
+import {
+  SkyDatepickerDate
+} from './datepicker-date';
+
+import {
+  SkyDatepickerService
+} from './datepicker.service';
+
+/**
+ * @internal
+ */
+@Component({
+  selector: 'sky-daypicker-cell',
+  templateUrl: 'daypicker-cell.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SkyDayPickerCellComponent implements OnInit, OnDestroy {
+
+  /**
+   * Specifies if the active date has been changed.
+   */
+  @Input()
+  public activeDateHasChanged: boolean;
+
+  /**
+   * Specifies the date this picker cell will represent on the calendar.
+   */
+  @Input()
+  public date: SkyDatepickerDate;
+
+  public hasTooltip: boolean = false;
+
+  public popoverController = new Subject<SkyPopoverMessage>();
+
+  private activeUid: string = '';
+
+  private cancelPopover: boolean = false;
+
+  private popoverOpen: boolean = false;
+
+  private ngUnsubscribe = new Subject<void>();
+
+  public constructor(
+    private datepicker: SkyDatepickerCalendarInnerComponent,
+    private datepickerService: SkyDatepickerService
+  ) {}
+
+  public ngOnInit(): void {
+    this.hasTooltip =
+      this.date.keyDate &&
+      this.date.keyDateText &&
+      this.date.keyDateText.length > 0 &&
+      this.date.keyDateText[0].length > 0;
+
+    // show the tooltip if this is the active date and is not the
+    // initial active date (activeDateHasChanged)
+    if (
+      this.datepicker.isActive(this.date) &&
+      this.activeDateHasChanged &&
+      this.hasTooltip
+    ) {
+      this.activeUid = this.date.uid;
+      this.showTooltip();
+    }
+
+    if (this.hasTooltip) {
+      this.datepickerService.keyDatePopoverStream
```

Comment:
I don't see `datepickerService.keyDatePopoverStream` referenced anywhere besides this file. Could it be moved internally to this file?

Reply:
That's true, but there are multiple daypicker cells on each calendar. This message stream uses the `SkyDatepickerService` to communicate to all the daypicker cells in the DOM to make sure there is only one popup displayed at anytime.